### DISCUSSION TITLE
Better way to handle HTML templates combined with translations

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#14631] [Features and ideas]
Currently we have some HTML snippets in language files, which may be okay for smaller HTML snippets.
However it is also used for larger HTML templates like in user invite flow: https://github.com/umbraco/Umbraco-CMS/blob/b923c3252537468a7ecad2d394f37c0dbc7387ae/src/Umbraco.Core/EmbeddedResources/Lang/en.xml#L2149

but then in Danish translation file it isn't consistent.
https://github.com/umbraco/Umbraco-CMS/blob/b923c3252537468a7ecad2d394f37c0dbc7387ae/src/Umbraco.Core/EmbeddedResources/Lang/da.xml#L1919-L1920

Also if any changes are made to the HTML document (in this case an email template), it need to be adjusted in each language file.

Could we instead use static HTML templates and static embed resource or maybe razor template like in Vendr / Umbraco Commerce?
Then the translation files are responsible to translate the actual text parts either using dictionary items or replacing the template variables (as known from different newsletter systems).

**Comment by bergmania** [comment]
There's definitely pros and cons to both solutions. The current solution allows very different HTML to be used for the languages, but I also see the issue when you only wanna do pure translations.

It would be cool with someway to override the mail bodies with razor templates.

**Reply by bjarnef** [reply]
Yes, I think there may be use-cases where it makes sense to have a different template per culture, but I think in most projects (and by default) it would just translate the text/content.

If it uses static HTML files, maybe it would be possible to override this per culture via a registered file in App_Plugins folder, e.g. `~/App_Plugins/MyPlugin/Views/emails/email-template_da-DK.html` similar to the core language files can be overridden.

https://docs.umbraco.com/umbraco-cms/extending/language-files#package-language-files

But maybe something with razor and how @mattbrailsford handle this in Vendr / Umbraco Commerce would we cool and standardize it so core, forms, commerce can benefits from this?
